### PR TITLE
RevertUnchanged on Submit option added

### PR DESCRIPTION
**Pull request by JustinasPl**

### CODE REVIEW

File: Common/Commands/Config.h [hunk 1]
```diff
@@ -127,7 +127,8 @@ class ConfigResponse : public BaseResponse
 				(requiresNetwork ? 1 : 0) + 
 				(enablesCheckout ? 1 : 0) + 
 				(enablesLocking  ? 1 : 0) +
-				(enablesRevertUnchanged  ? 1 : 0);
+				(enablesRevertUnchanged  ? 1 : 0)+
+                (enablesRevertUnchangedOnSubmit ? 1 : 0);
```
Comment (aras-p) [inline Common/Commands/Config.h]: Strange indentation everywhere. Looks like previous source files used actual tabs for indentation, and yours is using spaces.

**Comment by RobJessop** [comment]
Hi. What is this for? Reverting unchanged files on submit is already a Perforce Workspace option - is this for Plastic?

e.g. 
image

**Comment by aras-p** [comment]
@RobJessop this was a request from A2 folks. Apparently people forget / don't setup the workspaces correctly with this option, and they'd like that option to be in project settings somewhere. That said, yeah maybe instead we should just educate people and write better "how to setup perforce effectively" docs.

**Comment by RobJessop** [comment]
This particular case could be solved by A2 making revert unchanged the default on new workspaces. There are server side hooks for modifying forms (e.g. the client form in this case) to set a default and for validating forms on submit to enforce rules. See https://www.perforce.com/manuals/v15.1/p4sag/chapter.scripting.html (see "Triggering on forms" here: https://www.perforce.com/manuals/v15.1/p4sag/chapter.scripting.html).

This particular change looks like it will add value to PlasticSCM, but I don't think that generally we should be adding Unity features and risk to compensate for Perforce features being opaque.

**Comment by aras-p** [comment]
Ok makes sense. Maybe let's just not do the actual code change, and when we'll actually get around to write "how to perforce" docs for manual let's mention all the triggers and settings. Deal?

**Comment by RobJessop** [comment]
Yeah, this seems like one of those "Perforce is weird, ancient, possibly evil and needs a lot of work translating it for people to actually use" cases. Jake and Lee (Hammerton) did a lot of Perforce admin at our last company and would also be able to help with a guide for A2.

**Comment by aras-p** [comment]
Decided to not do this now, and document best practices instead

**Pull request closed without merging**